mono/metadata/runtime.c

```diff
@@ -54,32 +54,104 @@ mono_runtime_is_shutting_down (void)
 static void
 fire_process_exit_event (MonoDomain *domain, gpointer user_data)
 {
-	MonoError error;
-	MonoClassField *field;
-	gpointer pa [2];
-	MonoObject *delegate, *exc;
-
-	field = mono_class_get_field_from_name (mono_defaults.appdomain_class, "ProcessExit");
-	g_assert (field);
+	MonoMethod * method = mono_class_get_method_from_name_flags (
+		mono_defaults.appdomain_class, "QueueProcessExitEvent", 
+		0, METHOD_ATTRIBUTE_PRIVATE
+	);
 
-	delegate = *(MonoObject **)(((char *)domain->domain) + field->offset);
-	if (delegate == NULL)
+	// FIXME: The assert causes a crash during make... maybe because mscorlib is old?
+	if (!method)
```

Comment:
There should be only the assert bellow

Reply:
How do I successfully build with the assert there?

Reply:
Bump this https://github.com/mono/mono/blob/master/configure.ac#L43 and the build will take care of it